test/common/router/shadow_writer_impl_test.cc

```diff
@@ -53,18 +56,26 @@ class ShadowWriterImplTest : public testing::Test {
 TEST_F(ShadowWriterImplTest, Success) {
   InSequence s;
 
-  expectShadowWriter("cluster1", "cluster1-shadow");
+  expectShadowWriter("cluster1", "cluster1-shadow", false);
   Http::ResponseMessagePtr response(new Http::ResponseMessageImpl());
   callback_->onSuccess(request_, std::move(response));
 }
 
 TEST_F(ShadowWriterImplTest, Failure) {
   InSequence s;
 
-  expectShadowWriter("cluster1:8000", "cluster1-shadow:8000");
+  expectShadowWriter("cluster1:8000", "cluster1-shadow:8000", false);
   callback_->onFailure(request_, Http::AsyncClient::FailureReason::Reset);
 }
 
+TEST_F(ShadowWriterImplTest, SuccessNoAppend) {
```

Comment:
nit: change the name of the test to something more understandable, e.g., SuccessAvoidShadowHeaderPrefixChange.
Also feel free to add a one-line comment that describes the test.